spec/support/shared_examples_for_object_equality.rb

```diff
@@ -0,0 +1,59 @@
+
+# Examples for making sure two objects are equal
+shared_examples_for "object_equality" do
+  let(:not_equal_to_anything_else) { TestUtils::NotEqualToAnythingElse.new }
+
+  it "should find the first object to be equal to the second object" do
+    expect(first_object).to eq(second_object)
+  end
+
+  it "should find the second object to be equal to the first object" do
+    expect(second_object).to eq(first_object)
+  end
+
+  it "should the hash codes of the two objects to be equal" do
+    expect(first_object.hash).to eq(second_object.hash)
+  end
+
+  it "should find the first object not equal to a random other thing" do
```

Comment:
The two examples below are repeated across both of these shared examples, so I'm thinking maybe they should be pulled out into their own set of shared examples (I'm fairly certain you can include shared examples from another set of shared examples).